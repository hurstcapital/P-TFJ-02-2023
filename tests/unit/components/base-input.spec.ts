import { shallowMount } from "@vue/test-utils";
import BaseInputVue from "@/components/common/BaseInput.vue";

describe("BaseInput.vue", () => {
  it("should renders placeholder", () => {
    const placeHolder = "Place holder example";

    const wrapper = shallowMount(BaseInputVue, {
      props: {
        placeHolder: placeHolder,
      },
    });

    expect(wrapper.find("input").attributes("placeholder")).toMatch(
      placeHolder
    );
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should update v-model", async () => {
    const text = "text example";

    const wrapper = shallowMount(BaseInputVue, {
      props: {
        modelValue: "initialText",
        "onUpdate:modelValue": (e: Event) =>
          wrapper.setProps({ modelValue: e }),
      },
    });

    await wrapper.find("input").setValue(text);

    expect(wrapper.props("modelValue")).toBe(text);
    expect(wrapper.element).toMatchSnapshot();
  });
});
