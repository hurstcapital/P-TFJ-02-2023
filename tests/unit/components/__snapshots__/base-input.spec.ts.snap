// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseInput.vue should renders placeholder 1`] = `
<input
  class="w-full focus:outline-none m-1 ml-3"
  data-cy="input"
  placeholder="Place holder example"
  type="text"
/>
`;

exports[`BaseInput.vue should update v-model 1`] = `
<input
  class="w-full focus:outline-none m-1 ml-3"
  data-cy="input"
  placeholder=""
  type="text"
/>
`;
